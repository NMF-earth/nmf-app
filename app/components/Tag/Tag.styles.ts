import { StyleSheet, ViewStyle } from "react-native";

import { Colors } from "style";

const container: ViewStyle = {
  paddingVertical: 6,
  alignItems: "center",
  paddingHorizontal: 12,
  marginRight: 12,
  borderWidth: 1,
  borderRadius: 8,
};

export default StyleSheet.create({
  containerNotSelected: {
    ...container,
    backgroundColor: Colors.green50 + "10",
    borderColor: Colors.grey100 + 20,
  },
  containerSelected: {
    ...container,
    backgroundColor: Colors.green50 + "20",
    borderColor: Colors.green50,
  },
  containerIcon: {
    minWidth: 110,
    paddingVertical: 10,
  },
  icon: {
    marginBottom: 2,
  },
});
