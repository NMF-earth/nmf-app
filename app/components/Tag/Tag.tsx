import React from "react";
import { TouchableOpacity, View } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import { Colors } from "style";

import Text from "../Text";
import styles from "./Tag.styles";

interface Props {
  icon?: string;
  text: string;
  onPress: () => void;
}

const Tag: React.FC<Props> = ({ text, onPress, icon }) => {
  let iconItem = null;

  if (icon) {
    iconItem = (
      <Ionicons
        name={icon as keyof typeof Ionicons.glyphMap}
        size={32}
        style={styles.mainIcon}
        color={Colors.green50}
      />
    );
  }

  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      {iconItem}
      <View style={styles.textContainer}>
        <Text.Primary style={styles.text}>{text}</Text.Primary>
      </View>
      <Ionicons name={"chevron-forward"} size={20} color={Colors.green50} />
    </TouchableOpacity>
  );
};

export default Tag;
