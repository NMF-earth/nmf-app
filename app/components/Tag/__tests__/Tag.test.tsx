import React from "react";
import { create } from "react-test-renderer";

import Tag from "../Tag";

jest.unmock("../Tag.tsx");

let props;

beforeEach(() => {
  props = {
    selected: false,
    title: "Transport",
    onPress: () => {
      // do nothing.
    },
  };
});

/* TEXT ONLY */

it("Tag renders correctly with no icon", () => {
  const tree = create(<Tag {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with no icon and selected", () => {
  const tree = create(<Tag {...props} selected />).toJSON();
  expect(tree).toMatchSnapshot();
});

/* WITH ICON */

it("Tag renders correctly with airplane icon", () => {
  const tree = create(<Tag {...props} icon="md-airplane" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with airplane icon and selected", () => {
  const tree = create(<Tag {...props} icon="md-airplane" selected />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with restaurant icon", () => {
  const tree = create(<Tag {...props} icon="md-restaurant" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with restaurant icon and selected", () => {
  const tree = create(
    <Tag {...props} icon="md-restaurant" selected />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with build icon", () => {
  const tree = create(<Tag {...props} icon="md-build" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with build icon and selected", () => {
  const tree = create(<Tag {...props} icon="md-build" selected />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with md-play-circle icon", () => {
  const tree = create(<Tag {...props} icon="md-play-circle" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with md-play-circle icon and selected", () => {
  const tree = create(
    <Tag {...props} icon="md-play-circle" selected />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with md-flash icon", () => {
  const tree = create(<Tag {...props} icon="md-flash" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Tag renders correctly with md-flash icon and selected", () => {
  const tree = create(<Tag {...props} icon="md-flash" selected />).toJSON();
  expect(tree).toMatchSnapshot();
});
