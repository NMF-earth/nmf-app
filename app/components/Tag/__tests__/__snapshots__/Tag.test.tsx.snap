// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly with airplane icon 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0F853B20",
      "borderColor": "#0F853B",
      "borderRadius": 8,
      "borderWidth": 2,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 8,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <Ionicons
    color="#0F853B"
    name="md-airplane"
    size={32}
    style={
      {
        "marginRight": 16,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text.Primary
      style={
        {
          "textTransform": "capitalize",
        }
      }
    >
      Transport
    </Text.Primary>
  </View>
  <Ionicons
    color="#0F853B"
    name="chevron-forward"
    size={20}
  />
</TouchableOpacity>
`;

exports[`Tag renders correctly with no icon 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0F853B20",
      "borderColor": "#0F853B",
      "borderRadius": 8,
      "borderWidth": 2,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 8,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text.Primary
      style={
        {
          "textTransform": "capitalize",
        }
      }
    >
      Transport
    </Text.Primary>
  </View>
  <Ionicons
    color="#0F853B"
    name="chevron-forward"
    size={20}
  />
</TouchableOpacity>
`;
